File: codeup/100000572/E.cpp
#include <iostream>
using namespace std;

struct student {
  int num;
  char name[20];
  int score1;
  int score2;
  int score3;
};

int main() {
  student list[10];
  int score1_sum, score2_sum, score3_sum, score_max_iter;
  score1_sum = score2_sum = score3_sum = score_max_iter = 0;

  for (int i = 0; i < 10; i++) {
    scanf("%d %s %d %d %d", &list[i].num, list[i].name, &list[i].score1,
          &list[i].score2, &list[i].score3);
    score1_sum += list[i].score1;
    score2_sum += list[i].score2;
    score3_sum += list[i].score3;
    if (list[score_max_iter].score1 + list[score_max_iter].score2 +
            list[score_max_iter].score3 <
        list[i].score1 + list[i].score2 + list[i].score3)
      score_max_iter = i;
  }

  printf("%.2lf %.2lf %.2lf\n", score1_sum / 10.0, score2_sum / 10.0,
         score3_sum / 10.0);
  printf("%d %s %d %d %d\n", list[score_max_iter].num,
         list[score_max_iter].name, list[score_max_iter].score1,
         list[score_max_iter].score2, list[score_max_iter].score3);

  return 0;
}

File: codeup/100000572/E.py
#!/usr/bin/python3

class student:
    def __init__(self, num, name, score1, score2, score3):
        self.num = num
        self.name = name
        self.score1 = score1
        self.score2 = score2
        self.score3 = score3

    def __str__(self):
        return "{} {} {} {} {}".format(self.num, self.name, self.score1, self.score2, self.score3)


student_list = []
score1_sum = score2_sum = score3_sum = score_max_iter = 0
for i in range(10):
    num, name, score1, score2, score3 = list(input().split())
    score1, score2, score3 = list(map(int, [score1, score2, score3]))
    score1_sum += score1
    score2_sum += score2
    score3_sum += score3
    student_list.append(student(num, name, score1, score2, score3))
    score_sum = student_list[score_max_iter].score1 + \
        student_list[score_max_iter].score2+student_list[score_max_iter].score3
    if score_sum < score1+score2+score3:
        score_max_iter = i

print("{:.2f} {:.2f} {:.2f}".format(
    score1_sum/10, score2_sum/10, score3_sum/10))
print(student_list[score_max_iter])
